From 22693376144cbbe2547f60a6aa48fb56cf8247d1 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Thu, 14 Sep 2017 11:07:41 +0800
Subject: [PATCH 112/256] ARM: at91: Add the SoC options to Kconfig

To prepare to remove the SoCs options such as SAMA5D2, SAMA5D3
and SAMA5D4 from the CONFIG_SYS_EXTRA_OPTIONS option which is
deprecated, add the SoC options to Kconfig.

Signed-off-by: Wenyou Yang <wenyou.yang@microchip.com>
---
 arch/arm/mach-at91/Kconfig | 106 +++++++++++++++++++++++++++++++++------------
 1 file changed, 79 insertions(+), 27 deletions(-)

diff --git a/arch/arm/mach-at91/Kconfig b/arch/arm/mach-at91/Kconfig
index 513963c7da..0e71b69a19 100644
--- a/arch/arm/mach-at91/Kconfig
+++ b/arch/arm/mach-at91/Kconfig
@@ -3,6 +3,58 @@ if ARCH_AT91
 config AT91FAMILY
 	def_bool y
 
+config AT91SAM9260
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9G20
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9XE
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9261
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9263
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9G45
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9M10G45
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9N12
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9RL
+	bool
+	select CPU_ARM926EJS
+
+config AT91SAM9X5
+	bool
+	select CPU_ARM926EJS
+
+config SAMA5D2
+	bool
+	select CPU_V7
+
+config SAMA5D3
+	bool
+	select CPU_V7
+
+config SAMA5D4
+	bool
+	select CPU_V7
+
 choice
 	prompt "Atmel AT91 board select"
 	optional
@@ -13,24 +65,24 @@ config TARGET_AT91RM9200EK
 
 config TARGET_AT91SAM9260EK
 	bool "Atmel at91sam9260 reference board"
-	select CPU_ARM926EJS
+	select AT91SAM9260
 	select BOARD_EARLY_INIT_F
 
 config TARGET_ETHERNUT5
 	bool "Ethernut5 board"
-	select CPU_ARM926EJS
+	select AT91SAM9XE
 
 config TARGET_SNAPPER9260
 	bool "Support snapper9260"
-	select CPU_ARM926EJS
+	select AT91SAM9260
 	select DM
 	select DM_SERIAL
 	select DM_GPIO
 
 config TARGET_GURNARD
 	bool "Support gurnard"
+	select AT91SAM9G45
 	select BOARD_LATE_INIT
-	select CPU_ARM926EJS
 	select DM
 	select DM_SERIAL
 	select DM_GPIO
@@ -38,67 +90,67 @@ config TARGET_GURNARD
 
 config TARGET_AT91SAM9261EK
 	bool "Atmel at91sam9261 reference board"
-	select CPU_ARM926EJS
+	select AT91SAM9261
 	select BOARD_EARLY_INIT_F
 
 config TARGET_PM9261
 	bool "Ronetix pm9261 board"
-	select CPU_ARM926EJS
+	select AT91SAM9261
 
 config TARGET_AT91SAM9263EK
 	bool "Atmel at91sam9263 reference board"
-	select CPU_ARM926EJS
+	select AT91SAM9263
 	select BOARD_EARLY_INIT_F
 
 config TARGET_USB_A9263
 	bool "Caloa USB A9260 board"
-	select CPU_ARM926EJS
+	select AT91SAM9263
 
 config TARGET_PM9263
 	bool "Ronetix pm9263 board"
-	select CPU_ARM926EJS
+	select AT91SAM9263
 
 config TARGET_AT91SAM9M10G45EK
 	bool "Atmel AT91SAM9M10G45-EK board"
-	select CPU_ARM926EJS
+	select AT91SAM9M10G45
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_PM9G45
 	bool "Ronetix pm9g45 board"
-	select CPU_ARM926EJS
+	select AT91SAM9G45
 
 config TARGET_PICOSAM9G45
 	bool "Mini-box picosam9g45 board"
-	select CPU_ARM926EJS
+	select AT91SAM9M10G45
 	select SUPPORT_SPL
 
 config TARGET_AT91SAM9N12EK
 	bool "Atmel AT91SAM9N12-EK board"
-	select CPU_ARM926EJS
+	select AT91SAM9N12
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_AT91SAM9RLEK
 	bool "Atmel at91sam9rl reference board"
-	select CPU_ARM926EJS
+	select AT91SAM9RL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_AT91SAM9X5EK
 	bool "Atmel AT91SAM9X5-EK board"
-	select CPU_ARM926EJS
+	select AT91SAM9X5
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_SAMA5D2_PTC
 	bool "SAMA5D2 PTC board"
-	select CPU_V7
+	select SAMA5D2
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_SAMA5D2_XPLAINED
 	bool "SAMA5D2 Xplained board"
-	select CPU_V7
+	select SAMA5D2
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
@@ -117,41 +169,41 @@ config TARGET_SAMA5D27_SOM1_EK
 
 config TARGET_SAMA5D3_XPLAINED
 	bool "SAMA5D3 Xplained board"
-	select CPU_V7
+	select SAMA5D3
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_SAMA5D3XEK
 	bool "SAMA5D3X-EK board"
+	select SAMA5D3
 	select BOARD_LATE_INIT
-	select CPU_V7
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_SAMA5D4_XPLAINED
 	bool "SAMA5D4 Xplained board"
-	select CPU_V7
+	select SAMA5D4
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_SAMA5D4EK
 	bool "SAMA5D4 Evaluation Kit"
-	select CPU_V7
+	select SAMA5D4
 	select SUPPORT_SPL
 	select BOARD_EARLY_INIT_F
 
 config TARGET_MA5D4EVK
 	bool "Aries MA5D4EVK Evaluation Kit"
-	select CPU_V7
+	select SAMA5D4
 	select SUPPORT_SPL
 
 config TARGET_MEESC
 	bool "Support meesc"
-	select CPU_ARM926EJS
+	select AT91SAM9263
 
 config TARGET_CORVUS
 	bool "Support corvus"
-	select CPU_ARM926EJS
+	select AT91SAM9M10G45
 	select SUPPORT_SPL
 	select DM
 	select DM_SERIAL
@@ -160,7 +212,7 @@ config TARGET_CORVUS
 
 config TARGET_TAURUS
 	bool "Support taurus"
-	select CPU_ARM926EJS
+	select AT91SAM9G20
 	select SUPPORT_SPL
 	select DM
 	select DM_SERIAL
@@ -169,7 +221,7 @@ config TARGET_TAURUS
 
 config TARGET_SMARTWEB
 	bool "Support smartweb"
-	select CPU_ARM926EJS
+	select AT91SAM9260
 	select SUPPORT_SPL
 	select DM
 	select DM_SERIAL
@@ -178,7 +230,7 @@ config TARGET_SMARTWEB
 
 config TARGET_VINCO
 	bool "Support VINCO"
-	select CPU_V7
+	select SAMA5D4
 	select SUPPORT_SPL
 
 endchoice
-- 
2.13.6

